Tests/AEPOptimizeTests/UnitTests/OptimizePublicAPITests.swift

```diff
@@ -563,6 +560,455 @@ class OptimizePublicAPITests: XCTestCase {
+                              data: nil)
+
+        // test
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(type: testEvent.type,
+                                                                                    source: testEvent.source) { _ in
+            expectation.fulfill()
+        }
+
+        // test
+        Optimize.getPropositionsForSurfacePaths([""]) { _, _ in }
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+    }
+
+    func testGetPropositionsForSurfacePaths_validAndInvalidSurfaces() {
+        // setup
+        let expectation = XCTestExpectation(description: "getPropositions should dispatch an event with expected data.")
+        expectation.assertForOverFulfill = true
+
+        let testEventData: [String: Any] = [
+            "requesttype": "getpropositions",
+            "surfaces": [
+                "myView#htmlElement"
+            ]
+        ]
+        let testEvent = Event(name: "Optimize Get Propositions Request",
+                              type: "com.adobe.eventType.optimize",
+                              source: "com.adobe.eventSource.requestContent",
+                              data: testEventData)
+
+        // test
+        EventHub.shared.getExtensionContainer(MockExtension.self)?.registerListener(type: testEvent.type,
+                                                                                    source: testEvent.source) { event in
+            XCTAssertEqual(event.name, testEvent.name)
+            XCTAssertEqual("getpropositions", event.data?["requesttype"] as? String)
+            guard let surfaces: [String] = event.getTypedData(for: "surfaces") else {
+                XCTFail("Surface string array should be valid.")
+                return
+            }
+            XCTAssertEqual(1, surfaces.count)
+            XCTAssertEqual("myView#htmlElement", surfaces[0])
+            expectation.fulfill()
+        }
+
+        // test
+        Optimize.getPropositionsForSurfacePaths(["", "myView#htmlElement"]) { _, _ in }
+
+        // verify
+        wait(for: [expectation], timeout: 1)
+    }
+    
+    func testSetPropositionsHandler_validPropositionForSurface() {
+        // setup
+        Optimize.isPropositionsListenerRegistered = false
+        let expectation = XCTestExpectation(description: "setPropositionsHandler should be called with response event upon personalization notification.")
+        expectation.assertForOverFulfill = true
+
+        let testEvent = Event(name: "Personalization Notification",
+                              type: "com.adobe.eventType.optimize",
+                              source: "com.adobe.eventSource.notification",
+                              data: [
+                                "propositions": [
+                                    [
+                                        "id": "2cceff23-5eea-4bad-af5f-abb1aca1ea2e",
+                                        "scope": "myView#htmlElement",
```

Comment:
Should be the full prefixed surface path here

Reply:
That's correct.